cabal-testsuite/PackageTests/JS/JsOptions/jsoptions.cabal

```diff
@@ -0,0 +1,19 @@
+cabal-version: 3.14
+name: jsoptions
+version: 0
+build-type: Simple
+
+library
+  default-language: Haskell2010
+  exposed-modules: Lib
+  build-depends: base
+
+  if impl(javascript)
+    js-sources: jsbits/lib.js
+    js-options: -optJSP-DPRINT_DEF
```

Comment:
This is wrong. Now it's just a new name for `ghc-options`.

If `js-options` is introduced as a separate field, they should be prepended by `-optJSP` automatically (like `cpp-options` or `cxx-options` are prepended with appropriate `-opt` flag).

Also these are **jsp** options, i.e. JavaScript Preprocessor options. (c.f. `cc-options` and `cpp-options`).

Reply:
> This is wrong. Now it's just a new name for ghc-options.

If what I need was achievable through `ghc-options`, there would be no ticket!

Your suggestion for remame and prepend makes sense, I was merely following a hint of how to implement it. If there is a better way to make those options be passed rather than by separate field, let me know.

Reply:
> If what I need was achievable through ghc-options, there would be no ticket!

That's a different issue.

As @hsyl20 pointed on GHC tracker

>  So it looks like the ghc-options isn't passed by Cabal when building the js-sources

and **that's** the bug.

If there say `ghc-options: -Werror` is not passed by cabal when building `js-sources`, that's out right bug, which is not fixed by this PR.